### PR TITLE
[v14 - Ready] Fixes latexify tests

### PR DESCRIPTION
**Pull request by TorkelE**
Fixes the latexify tests. Mostly the errors were due to tabs.

There are a couple that we technically can generate passing tests for, however, the display is non-desired (for various reasons). For these I have just created a `test_broken` and linked the relevant issue.

The display of the tests that are marked as passing looks fine:
image
image
image
image
image
image
image
image

### CODE REVIEW

File: test/visualisation/latexify.jl [hunk 1]
```diff
@@ -48,68 +50,68 @@ let
         (d1,d2,d3,d4,d5,d6), (X1,X2,X3,X4,X5,X6)  ⟶ ∅
     end
 
-    # Latexify.@generate_test latexify(rn)
-    @test_broken latexify(rn; expand_functions = false) == replace(
```
Comment (TorkelE) [inline test/visualisation/latexify.jl:52]: just a simple error, `expand_functions = false` tested twice.
(`true` is the default, so don't really need it in the next one, but specified it anyway)

File: test/visualisation/latexify.jl [hunk 2]
```diff
@@ -48,68 +50,68 @@ let
...
-    \mathrm{X6} &\xrightarrow{d6} \varnothing  
-    \end{align*}
-    ", "\r\n"=>"\n")
-
-    # Latexify.@generate_test latexify(rn, mathjax=false)
-    @test_broken latexify(rn, mathjax = false) == replace(
-    raw"\begin{align*}
-    \varnothing &\xrightarrow{\frac{X4^{n1} v1^{2} K1^{n1}}{\left( K1^{n1} + X4^{n1} \right) \left( K1^{n1} + X2^{n1} \right)}} \mathrm{X1} \\
-    \varnothing &\xrightarrow{\frac{X5 v2}{K2 + X5}} \mathrm{X2} \\
-    \varnothing &\xrightarrow{\frac{K3 v3}{K3 + X3}} \mathrm{X3} \\
```
Comment (TorkelE) [inline test/visualisation/latexify.jl:98]: This just seemed to be an error, not sure what's up here.

File: test/visualisation/latexify.jl [hunk 3]
```diff
@@ -121,35 +123,49 @@ let
...
 let
     rn = @reaction_network begin
         p, 0 --> (m + n)*X
     end
-    
-    @test_broken latexify(rn) == replace(
-    raw"\begin{align*}
-    \varnothing &\xrightarrow{p} (m + n)\mathrm{X}  
-     \end{align*}
-    ", "\r\n"=>"\n")
```
Comment (TorkelE) [inline test/visualisation/latexify.jl:152]: this just seem to be git getting confused. No tests have been moved (although tabs and empty liens were removed in places, which i presume is causing this).

File: test/visualisation/latexify.jl [hunk 4]
```diff
@@ -170,21 +186,15 @@ let
...
-    $", "\r\n"=>"\n")
+    @test latexify(rn; form = :ode) == replace(
+raw"$\begin{align}
+\frac{\mathrm{d} X\left( t \right)}{\mathrm{d}t} =& p - d X\left( t \right) + 2 kD \mathrm{X2}\left( t \right) - \left( X\left( t \right) \right)^{2} kB \\
+\frac{\mathrm{d} \mathrm{X2}\left( t \right)}{\mathrm{d}t} =&  - kD \mathrm{X2}\left( t \right) + \frac{1}{2} \left( X\left( t \right) \right)^{2} kB
+\end{align}
+$", "\r\n"=>"\n")
+
+    # Currently latexify doesn't handle SDE systems properly, and they look identical to ode systems (https://github.com/SciML/ModelingToolkit.jl/issues/2782).
+    @test_broken false
```
Comment (TorkelE) [inline test/visualisation/latexify.jl:197]: `SDESystem` latex printing is broken.

File: test/visualisation/latexify.jl [hunk 5]
```diff
@@ -121,35 +123,49 @@ let
...
+    @test latexify(rn) == replace(
+raw"\begin{align*}
+\varnothing &\xrightarrow{p} (m + n)\mathrm{X}  
+ \end{align*}
+", "\r\n"=>"\n")
+end
+
+# Checks for systems with vector species/parameters.
+# Technically tests would work, however, the display is non-ideal (https://github.com/SciML/Catalyst.jl/issues/932, https://github.com/JuliaSymbolics/Symbolics.jl/issues/1167).
+let
```
Comment (TorkelE) [inline test/visualisation/latexify.jl:160]: New test for vector stuff (don't work right now).

File: test/visualisation/latexify.jl [hunk 6]
```diff
@@ -229,14 +239,15 @@ let
     end
 
     # Latexify.@generate_test latexify(rn)
-    @test_broken latexify(rn) == replace(
-    raw"\begin{align*}
-    \varnothing &\xrightarrow{p} (m + n)\mathrm{X}  
-    \end{align*}
-    ", "\r\n"=>"\n")
```
Comment (TorkelE) [inline test/visualisation/latexify.jl:236]: This test seems to have used the wrong generated stuff? In source we got
```julia
let
    rn = @reaction_network begin
        k*Y, Y --> ∅
    end

    # Latexify.@generate_test latexify(rn)
    @test_broken latexify(rn) == replace(
    raw"\begin{align*}
    \varnothing &\xrightarrow{p} (m + n)\mathrm{X}  
    \end{align*}
    ", "\r\n"=>"\n")
end
```

File: test/visualisation/latexify.jl [hunk 7]
```diff
@@ -247,18 +258,8 @@ let
     extended = extend(decaying_rate, base_network)
 
     # Latexify.@generate_test latexify(extended)
-    @test_broken latexify(extended) == replace(
-    raw"\begin{align*}
-    \mathrm{X} &\xrightarrow{k r} \varnothing  
-    0 &= -1 - x\left( t \right)  
-    \end{align*}
-    ", "\r\n"=>"\n")
```
Comment (TorkelE) [inline test/visualisation/latexify.jl:255]: While we could generate tests they would point towards LaTeX code we wouldn't want, so I removed it and just marked as something broken.

**Review by isaacsas — APPROVED** [review]

**Pull request merged**